Mathlib/RingTheory/Valuation/ValExtension.lean

```diff
@@ -0,0 +1,164 @@
+
+2. the valuation where `p` is mapped to one.
+
+For the algebraic closure of `ℚ_[p]`, if we choose the valuation of `p` to be one, then the
+restriction of this valuation to `K` equals the second valuation, but is only equivalent to the
+first valuation. The flexibility of equivalence here allows us to develop theory for both cases
+without first determining the normalizations once and for all.
+
+## Main Definition
+
+* `IsValExtension vR vA` : The valuation `vA` on `A` is an extension of the valuation `vR` on `R`.
+
+## References
+
+* [Bourbaki, Nicolas. *Commutative algebra*] Chapter VI §3, Valuations.
+* <https://en.wikipedia.org/wiki/Valuation_(algebra)#Extension_of_valuations>
+
+## Tags
+Valuation, Extension of Valuations
+
+-/
+open Valuation
+
+variable {R K A ΓR ΓK ΓA : Type*} [CommRing R] [Field K] [Ring A]
+    [LinearOrderedCommMonoidWithZero ΓR] [LinearOrderedCommMonoidWithZero ΓK]
+    [LinearOrderedCommMonoidWithZero ΓA] [Algebra R A] [Algebra K A]
+    (vR : Valuation R ΓR) (vK : Valuation K ΓK) (vA : Valuation A ΓA)
+
+/--
+The class `IsValExtension R A` states that the valuation of `A` is an extension of the valuation
+on `R`. More precisely, the valuation on `R` is equivlent to the comap of the valuation on `A`.
+-/
+class IsValExtension : Prop where
+  /-- The valuation on `R` is equivlent to the comap of the valuation on `A` -/
+  val_isEquiv_comap : vR.IsEquiv <| vA.comap (algebraMap R A)
+
+namespace IsValExtension
+
+section CoeLemma
+
+variable [IsValExtension vR vA]
+
+-- @[simp] does not work because `vR` cannot be inferred from `R`.
+theorem val_map_le_iff (x y : R) : vA (algebraMap R A x) ≤ vA (algebraMap R A y) ↔ vR x ≤ vR y :=
+  val_isEquiv_comap.symm x y
+
+theorem val_map_lt_iff (x y : R) : vA (algebraMap R A x) < vA (algebraMap R A y) ↔ vR x < vR y := by
+  simpa only [not_le] using ((val_map_le_iff vR vA _ _).not)
+
+theorem val_map_eq_iff (x y : R) : vA (algebraMap R A x) = vA (algebraMap R A y) ↔ vR x = vR y :=
+  (IsEquiv.val_eq val_isEquiv_comap).symm
+
+theorem val_map_le_one_iff (x : R) : vA (algebraMap R A x) ≤ 1 ↔ vR x ≤ 1 := by
+  simpa only [_root_.map_one] using val_map_le_iff vR vA x 1
+
+theorem val_map_lt_one_iff (x : R) : vA (algebraMap R A x) < 1 ↔ vR x < 1 := by
+  simpa only [_root_.map_one, not_le] using (val_map_le_iff vR vA 1 x).not
+
+theorem val_map_eq_one_iff (x : R) : vA (algebraMap R A x) = 1 ↔ vR x = 1 := by
+  simpa only [le_antisymm_iff, _root_.map_one] using
+    and_congr (val_map_le_iff vR vA x 1) (val_map_le_iff vR vA 1 x)
+
+instance id : IsValExtension vR vR where
+  val_isEquiv_comap := by
+    simp only [Algebra.id.map_eq_id, comap_id]
+    rfl
+
+end CoeLemma
+
+variable {ΓR ΓA ΓK: Type*} [LinearOrderedCommGroupWithZero ΓR] [LinearOrderedCommGroupWithZero ΓK]
+    [LinearOrderedCommGroupWithZero ΓA] {vR : Valuation R ΓR} {vK : Valuation K ΓK}
+    {vA : Valuation A ΓA} [IsValExtension vR vA] [IsValExtension vK vA]
+
+section mk'
```

Comment:
Not sure this is the best name for the section, but given its length you can skip the sectioning, perhaps. Or did you want to insert the variables inside it?

Reply:
I'll reorganise the sections. The name `mk'` is really poor.